tests/callbacks/test_write_log_callback.py

```diff
@@ -0,0 +1,159 @@
+# pyright: reportMissingParameterType=false
+# pyright: reportCallIssue=false
+
+from pathlib import Path
+from unittest.mock import call, mock_open, patch
+
+import pytest
+from event_model import DataKey, Event, EventDescriptor, RunStart, RunStop
+
+from ibex_bluesky_core.callbacks.file_logger import HumanReadableFileCallback
+
+save_path = Path("C:\\") / "instrument" / "var" / "logs" / "bluesky" / "output_files"
+
+
+@pytest.fixture
+def cb() -> HumanReadableFileCallback:
+    return HumanReadableFileCallback(save_path, ["block", "dae"])
+
+
+def test_header_data_all_available_on_start(cb):
+    time = 1728049423.5860472
+    uid = "test123"
+    scan_id = 1234
+    run_start = RunStart(time=time, uid=uid, scan_id=scan_id)
+    with patch("ibex_bluesky_core.callbacks.file_logger.open", mock_open()) as mock_file:
+        cb.start(run_start)
+        result = save_path / f"{run_start['uid']}.txt"
+
+    mock_file.assert_called_with(result, "a")
+    # assert not mock_file().write.assert_any_call(f"scan_id: {scan_id}")
+    # time should have been renamed to start_time and converted to human readable
+    mock_file().write.assert_any_call("start_time: 2024-10-04 14:43:43\n")
+
+    mock_file().write.assert_any_call(f"uid: {uid}\n")
+
+
+def test_descriptor_data_does_nothing_if_doc_not_called_primary(cb):
+    desc = EventDescriptor(
+        data_keys={}, uid="someuid", time=123.4, run_start="n/a", name="notprimary"
+    )
+    cb.descriptor(desc)
+
+    assert desc["uid"] not in cb.descriptors
+
+
+def test_descriptor_adds_descriptor_if_name_primary(cb):
+    desc = EventDescriptor(data_keys={}, uid="someuid", time=123.4, run_start="n/a", name="primary")
+
+    cb.descriptor(desc)
+    assert desc["uid"] in cb.descriptors
+    assert cb.descriptors[desc["uid"]] == desc
+
+
+def test_event_prints_header_with_units_and_respects_precision_of_value_on_first_message():
+    field_name = "test"
+    cb = HumanReadableFileCallback(save_path, [field_name])
+    # This actually contains the precision
+    expected_value = 1.2345
+    units = "mm"
+    prec = 4
```

Comment:
Since `expected_value` is only to 4 d.p. we're not actually testing it gets formatted to 4 d.p. as far as I can tell?